core/src/main/java/org/apache/accumulo/core/client/admin/NewTableConfiguration.java

```diff
@@ -184,6 +186,7 @@ public NewTableConfiguration setLocalityGroups(Map<String,Set<Text>> groups) {
       String value = LocalityGroupUtil.encodeColumnFamilies(colFams);
       tmp.put(Property.TABLE_LOCALITY_GROUP_PREFIX + entry.getKey(), value);
     }
+    checkDisjoint(properties, tmp, "locality groups");
```

Comment:
why not do this check after the put below?

Reply:
Since checkDisjoint is checking the keys rather than the values for disjointness, performing the check at this point allowed any errors to be discovered before performing the join on map values. Either location should suffice, this was the earliest opportunity to perform the check.It won't bother me if you prefer it to be located after the next put. If so, let me know and I'll update the location.

Reply:
I think if someone calls `ntc.setProperties(Property.TABLE_LOCALITY_GROUPS.getKey()='foo')` (not java syntax) and then calls `ntc.setLocalityGroups(...)`  it will not detect that overlap unless checked after.

Reply:
Gotcha! Thanks